go/pkg/client/cas.go

```diff
@@ -881,13 +894,14 @@ func NewCompressedWriteBuffer(w io.Writer) (io.WriteCloser, chan error, error) {
 		// separate thread. As such, we also need a way to signal the main
 		// thread that the decoding has finished - which will have some delay
 		// from the last Write call.
-		_, err := decoder.WriteTo(w)
+		_, err := decoderW.Decoder.WriteTo(w)
```

Comment:
I think this could just be `decoderW.WriteTo(w)` since the Decoder is embedded. And similarly for the `decoderW.Decoder.Reset` call further up.

Reply:
Fixed, thanks!